packages/serverless/src/gcpfunction/events.ts

```diff
@@ -21,15 +21,17 @@ export function wrapEventFunction(
 }
 
 /** */
-function _wrapEventFunction(
-  fn: EventFunction | EventFunctionWithCallback,
+function _wrapEventFunction<F extends EventFunction | EventFunctionWithCallback>(
+  fn: F,
   wrapOptions: Partial<EventFunctionWrapperOptions> = {},
-): EventFunctionWithCallback {
+): (...args: Parameters<F>) => ReturnType<F> | Promise<void> {
```

Comment:
Unfortunately we cannot just do `F` here since it produces some weird TS error I don't know how to fix.